src/cascadia/TerminalConnection/TelnetConnection.cpp

```diff
@@ -0,0 +1,265 @@
+        // Each connection needs to make sure to drain the output from its backing host.
+        _hOutputThread.reset(CreateThread(nullptr,
+                                          0,
+                                          StaticOutputThreadProc,
+                                          this,
+                                          0,
+                                          nullptr));
+
+        THROW_LAST_ERROR_IF_NULL(_hOutputThread);
+
+        _transitionToState(ConnectionState::Connecting);
+    }
+
+    // Method description:
+    // - ascribes to the ITerminalConnection interface
+    // - handles the different possible inputs in the different states
+    // Arguments:
+    // the user's input
+    void TelnetConnection::WriteInput(hstring const& data)
+    {
+        if (!_isStateOneOf(ConnectionState::Connected, ConnectionState::Connecting))
+        {
+            return;
+        }
+
+        auto str = winrt::to_string(data);
+        if (str.size() == 1 && str.at(0) == L'\r')
+        {
+            str = "\r\n";
+        }
+
+        telnetpp::bytes bytes(reinterpret_cast<const uint8_t*>(str.data()), str.size());
+        _session.send(bytes, [=](telnetpp::bytes data) {
+            _socketSend(data);
+        });
+    }
+
+    // Method description:
+    // - ascribes to the ITerminalConnection interface
+    // - resizes the terminal
+    // Arguments:
+    // - the new rows/cols values
+    void TelnetConnection::Resize(uint32_t /*rows*/, uint32_t /*columns*/)
+    {
+        if (!_isConnected())
+        {
+            /*_initialRows = rows;
+            _initialCols = columns;*/
+        }
+        else
+        {
+        }
+    }
+
+    // Method description:
+    // - ascribes to the ITerminalConnection interface
+    // - closes the websocket connection and the output thread
+    void TelnetConnection::Close()
+    {
+        try
+        {
+            if (_transitionToState(ConnectionState::Closing))
+            {
+                _socket.Close();
+                if (_hOutputThread)
+                {
+                    // Tear down our output thread
+                    WaitForSingleObject(_hOutputThread.get(), INFINITE);
+                    _hOutputThread.reset();
+                }
+
+                _transitionToState(ConnectionState::Closed);
+            }
+        }
+        catch (...)
+        {
+            LOG_CAUGHT_EXCEPTION();
+            _transitionToState(ConnectionState::Failed);
+        }
+    }
+
+    // Method description:
+    // - this method bridges the thread to the Azure connection instance
+    // Arguments:
+    // - lpParameter: the Azure connection parameter
+    // Return value:
+    // - the exit code of the thread
+    DWORD WINAPI TelnetConnection::StaticOutputThreadProc(LPVOID lpParameter)
+    {
+        auto pInstance = reinterpret_cast<TelnetConnection*>(lpParameter);
+        return pInstance->_outputThread();
+    }

```

Comment:
Tiny nit: this can be folded into the call to `CreateThread` using the method from `ConptyConnection`